### PR TITLE
tuntox 0.0.10.1 (new formula)

### PR DESCRIPTION
**Pull request by artenator**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/tuntox.rb
Comment (carlocab) [inline Formula/tuntox.rb:15]: Won't this lead to garbage in the version output?

File: Formula/tuntox.rb
Comment (carlocab) [inline Formula/tuntox.rb:16]:
```suggestion
      system "make", "-f", "Makefile.mac", "LIB_DIR=#{HOMEBREW_PREFIX}/lib"
```

**Review by carlocab — COMMENTED** [review]
This seems to statically link `libsodium` and `libtoxcore`, which isn't good.

**Comment by chenrui333** [comment]
ping @artenator

**Comment by artenator** [comment]
> This seems to statically link `libsodium` and `libtoxcore`, which isn't good.

Thanks for catching this. Looks like we can use `make tuntox_nostatic`, and the macos build is not static by default according to their documentation. I've added this change to the latest commit

File: Formula/tuntox.rb
Comment (artenator) [reply Formula/tuntox.rb:15]: Yes

The git sha is placed in `gitversion.h`

Would a simple hardcoded sha via an `inreplace` be sufficient here?

Comment (carlocab) [reply Formula/tuntox.rb:17]: Let's just replace it with `"N/A"` to avoid needing a git clone and also having to manually modify the commit on version bumps.

**Comment by carlocab** [comment]
> the macos build is not static by default according to their documentation.

This isn't the case, unfortunately. In `Makefile.mac`:

```make
tuntox: $(OBJECTS) $(INCLUDES)
	$(CC) -o $@ $(OBJECTS) -ltoxcore $(LDFLAGS) $(LIB_DIR)/libsodium.a $(LIB_DIR)/libtoxcore.a
```

https://github.com/gjedeer/tuntox/blob/abe4bf14de00702f540be8522d4cac06601f61cd/Makefile.mac#L22

Let's try using the `Makefile` unconditionally. It looks like it works on macOS too, and will link dynamically with `tuntox_nostatic`.

File: Formula/tuntox.rb
Comment (carlocab) [inline Formula/tuntox.rb:27]:
```suggestion
    inreplace "gitversion.h", /.*/, '#define GITVERSION "N/A"'
    system "make", "prefix=#{prefix}", "install"
```
Let's try this.

Comment (artenator) [reply Formula/tuntox.rb]: I think we might still need to keep some of the `inreplace`s for the dependencies on the local git repo. It seems like maybe the `FORCE` target makes the git portion run.

Do you think this is fine?

`inreplace "gitversion.h: FORCE", "gitversion.h:"`

File: Formula/tuntox.rb
Comment (carlocab) [inline Formula/tuntox.rb:15]: Oops, I think we don't need `-lrt` on macOS.
```suggestion
    inreplace "Makefile" do |s|
      s.gsub! "gitversion.h: FORCE", ""
      s.gsub! "-lrt", "" if OS.mac?
    end
```

**Comment by carlocab** [comment]
I owe you another review -- will find the time to get back to this. Ping me if you don't hear from me in a few days.

**Comment by artenator** [comment]
Hi @carlocab , hope you're doing well. Do you think this is good to go? Please let me know if there's anything we need to adjust 😄

**Comment by carlocab** [comment]
Apologies for the delay -- this is very close to ready. I was just hoping to be able to upstream the `-lrt` change before we merge this. It should be fairly simple (don't use it when `uname` is `Darwin`). Care to take a stab at it?

**Comment by artenator** [comment]
> Apologies for the delay -- this is very close to ready. I was just hoping to be able to upstream the `-lrt` change before we merge this. It should be fairly simple (don't use `-lrt` when `uname` is `Darwin`). Care to take a stab at it?

Sounds good! I'll make a PR on the tuntox repo for this change and keep you posted. Thanks for your continued help and time for helping me get this formula into homebrew core! 😃

**Comment by carlocab** [comment]
Great; let me know if you get stuck. Thanks for working on this!

**Comment by artenator** [comment]
Hey @carlocab!

My PR was merged into tuntox. Before asking the owner/maintainer to make a new tag, I wanted to double check with you to make sure what I included covers what we need. Thanks so much

**Comment by carlocab** [comment]
I've had a look at your change and it looks good. Are you able to build `tuntox` after including your changes? You can probably try them out with `brew install --HEAD tuntox` from your PR branch.

File: Formula/tuntox.rb
Comment (carlocab) [inline Formula/tuntox.rb:17]: I'm :+1: to merge this if we add a comment here that the `-lrt` can be removed after `0.0.10.1`.

**Comment by artenator** [comment]
> I've had a look at your change and it looks good. Are you able to build `tuntox` after including your changes? You can probably try them out with `brew install --HEAD tuntox` from your PR branch.

@carlocab 
Just tried this, and it works!

```
❯ brew install --build-from-source --HEAD tuntox
==> Cloning https://github.com/gjedeer/tuntox.git
Updating /Users/arteebrahimi/Library/Caches/Homebrew/tuntox--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 555c3a1 include print_version_stdout declaration in main.c, conditionally add -lrt flag for non-macos
==> make prefix=/opt/homebrew/Cellar/tuntox/HEAD-555c3a1 install
🍺  /opt/homebrew/Cellar/tuntox/HEAD-555c3a1: 5 files, 142KB, built in 2 seconds
==> Running `brew cleanup tuntox`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).

❯ which tuntox
/opt/homebrew/bin/tuntox

❯ tuntox
2022-07-24 18:40:15: [INFO]	Tuntox built from git commit N/A, toxcore version 0.2.18
2022-07-24 18:40:15: [INFO]	Using 32730 for TCP relay port and 56910-56920 for UDP
2022-07-24 18:40:15: [WARNING]	File not found (/etc/tuntox/tox_save)
2022-07-24 18:40:15: [WARNING]	Could not open save file
2022-07-24 18:40:15: [INFO]	Could not find Tox bootstrap nodes. Using hardcoded.
2022-07-24 18:40:15: [WARNING]	Could not open save file
2022-07-24 18:40:15: [WARNING]	Shared secret authentication is not used - skilled attackers may connect to your tuntox server
2022-07-24 18:40:15: [INFO]	Using Tox ID: C2728E4A97F02277CE845EB346416EF11ABCA2706C132BFFC7BD33F62805535C568EAA19CD9D
```

`brew test tuntox` looks to work as well. 

I've added the comment regarding the `-lrt` flag.

**Review by carlocab — APPROVED** [review]
Thanks for your work @artenator!

**Pull request closed without merging**
